crates/arrow_tools/src/record_batch.rs

```diff
@@ -0,0 +1,267 @@
+/*
+Copyright 2024 The Spice.ai OSS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+use arrow::{
+    array::{
+        new_null_array, Array, Int32Array, Int64Array, LargeStringArray, RecordBatch, StringArray,
+        TimestampMicrosecondArray, TimestampMillisecondArray, TimestampSecondArray,
+    },
+    datatypes::{DataType, Field, SchemaRef, TimeUnit},
+};
+use snafu::prelude::*;
+use std::{num::TryFromIntError, sync::Arc};
+
+pub type Result<T, E = Error> = std::result::Result<T, E>;
+
+#[derive(Debug, Snafu)]
+pub enum Error {
+    #[snafu(display("Error converting record batch: {}", source))]
+    ConvertRecordBatch { source: arrow::error::ArrowError },
```

Comment:
Nitpick: I think error names should represent the failure
```suggestion
    UnableToConvertRecordBatch { source: arrow::error::ArrowError },
```

Reply:
Yeah, rust-analyzer suggested a common prefix is not preferred but I guess I can override the linter rule in this case.